domain/port/service/service.go

```diff
@@ -158,134 +168,49 @@ func (s *Service) UpdateUnitPorts(ctx context.Context, unitUUID string, openPort
 		return errors.Errorf("cannot update unit ports with conflict(s): %w", err)
 	}
 
-	// construct a map of all port ranges being closed to the endpoint they're
-	// being closed on. Except the wildcard endpoint.
-	closePortsToEndpointMap := make(map[network.PortRange]string)
-	for endpoint, endpointClosePorts := range closePorts {
-		for _, portRange := range endpointClosePorts {
-			closePortsToEndpointMap[portRange] = endpoint
-		}
-	}
-	// Ensure endpoints closed on the wildcard endpoint are not in the map.
-	for _, wildcardClosePorts := range closePorts[WildcardEndpoint] {
-		delete(closePortsToEndpointMap, wildcardClosePorts)
-	}
-
 	err = s.st.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
-		// Verify input port ranges do no conflict with any port ranges co-located
-		// with the unit.
-		colocatedOpened, err := s.st.GetColocatedOpenedPorts(ctx, unitUUID)
-		if err != nil {
-			return errors.Errorf("failed to get opened ports co-located with unit %s: %w", unitUUID, err)
-		}
-		err = verifyNoPortRangeConflicts(allInputPortRanges, colocatedOpened)
+		err := s.verifyNoColocatedPortRangeConflicts(ctx, unitUUID, allInputPortRanges)
```

Comment:
Use typed uuids and verify them at the service layer, this includes all methods in the service layer.